### DISCUSSION TITLE
Feature request: graphic tablet support (for drawing over the map)

### DISCUSSION THREAD

**Discussion by MayoBous** [freeplane/freeplane#593] [Ideas]
Hi! I'm a big fan of Freeplane and I find it really helpful! Yet there are feautres that I really wish were there.

I'm a graphic designer and I usually work more often with my stylus and graphic tablet rather than with a mouse. It could be a nice feature if one could, for example, circle a certain area and leave a fast, handwritten comment.

I also pictured some sort of sticky notes system, where you could actually sketch ideas. It would come in handy when conceptualizing a logo or a very rough storyboard.

I have very little to none coding skills, but I'm very fond of freeplane and a feature like those I have in mind would be a nice addition!

**Comment by euu2021** [comment]
> I also pictured some sort of sticky notes system, where you could actually sketch ideas. It would come in handy when conceptualizing a logo or a very rough storyboard.

I think this could be done with a script that does something like that:

- creates a floating node
- inserts a blank image (maybe, as inline image?)
- opens the image on an external program, where the user can draw
- after finishing the drawing, save and close
- the drawing will appear in FP (a bit of refresh is needed)

I tried doing the steps manually:
![javaw_j8MOm0oEGm](https://user-images.githubusercontent.com/77707706/187052853-7c483219-3b14-44a1-9e82-42ee1d8b5402.gif)


---

>  It could be a nice feature if one could, for example, circle a certain area [...].

Maybe using two invisible floating nodes connected (with a connector). The connector can be used to make lines and arrows. A circle is a bit awkward to perform, but it can be done:

![javaw_MuGkWO0wSb](https://user-images.githubusercontent.com/77707706/187053107-184147f0-b23f-4ffb-b5f8-07aeafd5d45c.gif)

**Comment by adamfreeplaneuser01** [comment]
> > I also pictured some sort of sticky notes system, where you could actually sketch ideas. It would come in handy when conceptualizing a logo or a very rough storyboard.
> 
> I think this could be done with a script that does something like that:
> 
>     * creates a floating node
> 
>     * inserts a blank image (maybe, as inline image?)
> 
>     * opens the image on an external program, where the user can draw
> 
>     * after finishing the drawing, save and close
> 
>     * the drawing will appear in FP (a bit of refresh is needed)
> 
> 
> I tried doing the steps manually: ![javaw_j8MOm0oEGm](https://user-images.githubusercontent.com/77707706/187052853-7c483219-3b14-44a1-9e82-42ee1d8b5402.gif) [ ![javaw_j8MOm0oEGm](https://user-images.githubusercontent.com/77707706/187052853-7c483219-3b14-44a1-9e82-42ee1d8b5402.gif) ](https://user-images.githubusercontent.com/77707706/187052853-7c483219-3b14-44a1-9e82-42ee1d8b5402.gif) [ ](https://user-images.githubusercontent.com/77707706/187052853-7c483219-3b14-44a1-9e82-42ee1d8b5402.gif)
> 
> > It could be a nice feature if one could, for example, circle a certain area [...].
> 
> Maybe using two invisible floating nodes connected (with a connector). The connector can be used to make lines and arrows. A circle is a bit awkward to perform, but it can be done:
> 
> ![javaw_MuGkWO0wSb](https://user-images.githubusercontent.com/77707706/187053107-184147f0-b23f-4ffb-b5f8-07aeafd5d45c.gif) [ ![javaw_MuGkWO0wSb](https://user-images.githubusercontent.com/77707706/187053107-184147f0-b23f-4ffb-b5f8-07aeafd5d45c.gif) ](https://user-images.githubusercontent.com/77707706/187053107-184147f0-b23f-4ffb-b5f8-07aeafd5d45c.gif) [ ](https://user-images.githubusercontent.com/77707706/187053107-184147f0-b23f-4ffb-b5f8-07aeafd5d45c.gif)

Hi! Are you able to share the script that you used to be able to add drawings to Freeplane? Or do you have a more detailed tutorial that I could follow? Thanks!

**Reply by euu2021** [reply]
In my example, I did the process manually. Basically, I 

- pasted a blank image, 
- right-clicked on it, 
- open in viewer (MS Paint)
- edit, save, and close MS Paint
- move a little bit the image to refresh in FP

**Comment by i-plasm** [comment]
I made a script called 'Sketcher' for quickly sketching drawings on a node in a external editor and having the changes refreshed in the mindmap. I've implemented some of the ideas discussed by you and @euu2021 . I hope it helps!

**(New script) Sketcher: quickly sketch drawings on a node**:  https://github.com/freeplane/freeplane/discussions/1496

![sketcher_23-10-26](https://github.com/freeplane/freeplane/assets/144830522/572ef726-e782-4dc1-be5f-8abf4cf47c47)

**Comment by MayoBous** [comment]
It looks fantastic! Can't wait to try it out. Thank youuu!

**Comment by beccon4** [comment]
That is a really nice feature - but to appreciate it an basic integration of graphics into Freeplane is necessary. I mean without the need of installing plugins and without the need of sending more than one file when moving the map to someone else (who might have a different file structure and/or OS)  I know that this is not so popular - but this is the big, big showstopper to use Freeplane as knowledge base and dokumentation in the informal way of communicating (messengers - send over quickly - oops all images missing) we have nowadays.
